### ISSUE TITLE
JOSS review: Considering switching documentation to another branch . Something like gh-pages 

### ISSUE THREAD

**Issue by kanishkan91** [floodlight-sports/floodlight#87]
I see the docs are hosted on each branch. This would create problems for future development. E.g every time you wanted to update the documentation, you would have to re-merge into master etc. I would recommend you store the docs on a  separate branch. That way when you just need to update the documentation, you just do it on that branch and don't touch master at all. You can also link the new branch for documentation in your settings menu. See example below, 

![image](https://user-images.githubusercontent.com/37234086/183258272-04c3692f-9919-4114-807a-463348d529f1.png)

https://github.com/openjournals/joss-reviews/issues/4588

**Comment by draabe** [comment]
@kanishkan91 thanks for the suggestion! We've considered your proposal thoroughly, and though we realize that outsourcing the documentation to a separate branch would allow leaner updates of the documentation, we still think that this benefit does not match the associated difficulties in our case. More precisely, we think that the following points can be problematic if the documentation was on a different branch:

- We use (a simplified version of) the [git-flow branching model](https://nvie.com/posts/a-successful-git-branching-model/) and have had positive experiences using it. Especially for less experienced users, this model gives clear instructions on how to merge new contributions and the different stages until changes reach the master branch. Adding a separate documentation branch would partially conflict with this model, as there is not a single develop branch to consider for new releases, but a parallel documentation branch to keep in mind as well.
- Although small changes to the documentation would be quicker with a separate branch, we'd argue this is exactly the opposite for larger changes: if a contributor wants to merge a larger feature/fix that also modifies the corresponding documentation to the package, he needs to create two independent pull requests - one for the code, one for the docs, which could be much harder to maintain and keep both PRs in sync if changes had to be made before acceptance.
- More generally and most importantly, splitting the documentation into a new branch will have it lose touch of the code-base. Since the docs are closely related to the code, it probably makes sense to keep them together. By splitting, code and docs could diverge from each other leaving outdated documenation, either because the code changed or the docs haven't been updated.
- As it stands, we directly use the package content to create the docs. For example, to keep the version number up to date automatically (see ```\docs\source\conf.py```) or by using sphinx' ```autodoc``` to automatically generate module references from function docstrings. Thus, having the source on a different branch, all this functionality would break. There might be a workaround for this, but generally, this would require a refactor that is substantially larger than changing the branch for building the docs.
- We don't really see that updating the master branch (which is equivalent to a new release using the git-flow workflow) with "only" changes to documenation is an issue as you state. Especially since we follow Semantic Versioning, where changes in the documenation should also be represented by patch - this even explicitly requires a new merge into master. As we principally collect changes on the develop branch until a new release is scheduled (with a new merge into master), we also have the option to keep minor documentation changes on the develop branch until we feel it's time for a novel release, including documentation changes. If there's urgent documentation changes, a new patch is sensible anyways.

To sum up, we feel that separating the documentation on a new branch might be convenient, but could easily introduce more issues than it solves (as discussed above). As we use readthedocs (in comparison to GitHub pages), we also don't need a separate branch for serving.

Therefore, we'd prefer to keep the codebase and documentation together and cohesive, if that makes sense.

**Comment by crvernon** [comment]
@kanishkan91 @draabe this sounds reasonable to me.  Thank you!

https://github.com/openjournals/joss-reviews/issues/4588
